skypalwarmup/JSUserRegistrationService/src/test/java/application/controller/UserControllerTest.java

```diff
@@ -0,0 +1,112 @@
+package application.controller;
+
+import application.dao.UserRepository;
+import application.entity.AccountInformation;
+import application.entity.Contacts;
+import application.entity.PersonalDetails;
+import application.entity.User;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.data.domain.Example;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+/**
+ * Copyright (c) 2016-2018, Jumia.
+ */
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
```

Comment:
Line 30: If I'm not mistaken this will load the whole Spring context. If the intention here is to *Unit Test* the controller, and hence it makes sense to mock the `UserService` not the `UserRepository` (Otherwise you are testing more than one component), I think you could use the `@WebMvcTest(UserController.class)` instead.

Check out this link for support: http://www.baeldung.com/spring-boot-testing